src/gluonts/model/evaluation.py

```diff
@@ -0,0 +1,180 @@
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import logging
+from collections import ChainMap
+from typing import Iterable, Optional, Union
+from dataclasses import dataclass
+from toolz import first
+
+import numpy as np
+import pandas as pd
+
+from gluonts.dataset import DataEntry
+from gluonts.dataset.split import TestData
+from gluonts.time_feature.seasonality import get_seasonality
+from gluonts.model.forecast import Forecast
+from gluonts.ev import seasonal_error
+from gluonts.itertools import prod
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class BatchForecast:
+    """
+    Wrapper around ``Forecast`` objects, that adds a batch dimension
+    to arrays returned by ``__getitem__``, for compatibility with
+    ``gluonts.ev``.
+    """
+
+    forecast: Forecast
+    allow_nan: bool = False
+
+    def __getitem__(self, name):
+        value = self.forecast[name]
+        valid = ~np.isnan(value).any()
+
+        if not valid and self.allow_nan:
+            logger.warning(
+                "Forecast contains NaN values. Metrics may be incorrect."
+            )
+
+        if not valid and not self.allow_nan:
+            raise ValueError("Forecast contains NaN values")
+
+        return np.expand_dims(value, axis=0)
+
+
+def _get_data_batch(
+    input_: DataEntry,
+    label: DataEntry,
+    forecast: Forecast,
+    seasonality: Optional[int] = None,
+    mask_invalid_label: bool = True,
+    allow_nan_forecast: bool = False,
+) -> ChainMap:
+    forecast_dict = BatchForecast(forecast, allow_nan=allow_nan_forecast)
+
+    freq = forecast.start_date.freqstr
+    if seasonality is None:
+        seasonality = get_seasonality(freq=freq)
+
+    label_target = label["target"]
+    if mask_invalid_label:
+        label_target = np.ma.masked_invalid(label_target)
+
+    other_data = {
+        "label": np.expand_dims(label_target, axis=0),
+        "seasonal_error": np.expand_dims(
+            seasonal_error(
+                input_["target"], seasonality=seasonality, time_axis=-1
+            ),
+            axis=0,
+        ),
+    }
+
+    return ChainMap(other_data, forecast_dict)
+
+
+def evaluate(
+    metrics,
+    test_data: TestData,
+    forecasts: Iterable[Forecast],
+    *,
+    axis: Optional[Union[int, tuple]] = None,
+    mask_invalid_label: bool = True,
+    allow_nan_forecast: bool = False,
+    seasonality: Optional[int] = None
+) -> dict:
+    """
+    Evaluate ``metrics`` by comparing ``forecasts`` to ``test_data``.
+
+    The optional ``axis`` arguments controls aggregation of the metrics:
+    - ``None`` (default) aggregates across all dimensions
+    - ``0`` aggregates across the batch dimension
+    - ``1`` aggregates across the time dimension
+    - ``2`` aggregates across the target dimension
```

Comment:
Nit: I find the `target dimension` is a bit confusing. Is it for different quantiles or different time series in a multivariate setting or both? Might be worth to explain what does it mean in the docstring.

Reply:
It’s the dimension in the multivariate setting. But I’ll try to explain better how the `axis` dimension is to be used